awx/api/urls.py

```diff
@@ -411,6 +411,7 @@ def url(regex, view, kwargs=None, name=None, prefix=''):
     url(r'^hosts/(?P<pk>[0-9]+)/ansible_facts/$',             'host_ansible_facts_detail'),
     url(r'^jobs/(?P<pk>[0-9]+)/extra_credentials/$',          'job_extra_credentials_list'),
     url(r'^job_templates/(?P<pk>[0-9]+)/extra_credentials/$', 'job_template_extra_credentials_list'),
+    url(r'^user_session/$', 'user_session_view'),
```

Comment:
Note: the endpoint name is prone to change after token-based auth is purged.

Reply:
DRF itself contains views for handling session-based auth (this might be dependent on the work @wwitzel3 is doing upgrading dependencies):

http://www.django-rest-framework.org/tutorial/4-authentication-and-permissions/#adding-login-to-the-browsable-api

This also generates a login/logout button in the rest api.

`^api-auth/` given on that page (which is just an example btw) contain a /login and /logout view which I think can be referenced as the UI normally would. I'm quite certain that just accepts form-encoded username+password (on login) and should also support application/json.

Reply:
Good to know DRF has it's own view implementation. I will check to see if it can be tweaked a bit to be used by us. 

The plan, as I discussed with @rmfitzpatrick, was to finally substitute `/authtoken/` endpoint with `/user_session/` endpoint, meaning `/authtoken/` endpoint will be kept, but function just like `/user_session/` endpoint. Admittedly it's controversial, because `/authtoken/` will no longer function as 'providing/revoking auth tokens', but the purpose is to keep `/authtoken/` endpoint so no API v3 will be needed.

On the other hand, if we end up using `/api-auth/login/` and `/api-auth/logout/`, what should we do with `/authtoken/` endpoint? Should it be removed since it won't function any more? If it is removed does that mean an API v3 is needed?

Reply:
Auth tokens are going to need new end points that will be provided by oauth toolkit. We are going to remove our token support and leave basic auth for backwards compatibility.  Sessions need to use the drf built in views.

Reply:
That is to say that /authtoken will probably go away OR we might can reuse it for oauth... Though really I'd like to land everything under something like /auth

Reply:
The intended benefit of changing the implementation of `/authtoken` but largely keeping the current interface (post/delete) was that clients would only have to strip/update their response handling.  Adding a new endpoint for essentially the same purpose w/ sub login/logout endpoints seems overkill when a minor documentation detail is warranted imo.

`/authtoken` would not be used for oauth apps.

Reply:
*BUT* session auth is not at all related to dealing with tokens which will create confusion. I'd argue by revising the meaning of `/authtoken` you will be creating more confusion.

API only consumers won't care about session auth because they won't use it (or shouldn't). It'll be easier and more correct for them to use the oauth interface.  For the UI-side using session auth will not want to follow the POST/DELETE interface because it doesn't make any sense... they'll want to use the login/logout views provided by the DRF session interface because it's more idiomatic to their desired workflow.